client/components/Event/EventDetail.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import UpdateEventForm from './UpdateEventForm';
+import { Link } from 'react-router-dom';
+import { getEvents as _getEvents, getEventDetail as _getEventDetail, joinEvent as _joinEvent, unjoinEvent as _unjoinEvent } from '../../actions/events';
+import {
+    Button,
+    Container,
+    Grid,
+    Header,
+    Icon,
+    Image,
+    List,
+    Menu,
+    Segment,
+  } from 'semantic-ui-react';
+
+class EventDetail extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { 
+            going: false,
+            activeItem: 'info',
+            event: {}
+         }
+         this.rsvp = this.rsvp.bind(this);
+         this.unrsvp = this.unrsvp.bind(this);
+         
+    }
+
+    componentDidMount() {
+        this.props.getEventDetail(this.props.match.params.eventId);
+    }
+    
+    componentDidUpdate() {
+        const { event } = this.props;
+        const attendees = event.users;
+        
+
+        if (attendees && this.state.going === false) {
+            const isGoing = attendees.filter(user => user.id === this.props.user.id)
+            if (isGoing.length) {
+                this.setState({going: true})
+            }
+            }
+    }
+
+    rsvp() {
+        this.props.joinEvent(this.props.event)
+        this.setState({going: true})
+    }
+    unrsvp() {
+        this.props.unjoinEvent(this.props.event)
+        this.setState({going: false})
+    }
+
+    
+
+    handleMenuClick = (e, { name }) => this.setState({ activeItem: name })
+
+    render() { 
+        const { event, user, group } = this.props;
+        const { going, activeItem } = this.state;
+        const attendees = event.users;
+        console.log('EVENT', event)
+        console.log('history', this.props.history)
+        return ( 
+
+            <Fragment>
+                <Container stretch='true'>
+                <Segment style={{ padding: '8em 0em' }} vertical>
+                    { event.day ? 
+                    <Fragment>
+                    <Grid container stackable verticalAlign="middle" textAlign='center'>
+                        <Grid.Row>
+                            <Grid.Column>
+                                <Header as="h3" style={{ fontSize: '2em' }}>
+                                {event.name}
+                                </Header>
+                                
+                            </Grid.Column>
+                        </Grid.Row>
+                        <Grid.Row>
+                        { !going ? 
+                                        <Button onClick={this.rsvp} color='green'>I'm Going!</Button> :
+                                        <Button onClick={this.unrsvp} color='red'>Not Going</Button>
+                                    }
+                        <Menu tabular attached='top'>
```

Comment:
There is some really interesting formatting going on here with { !going }
Wonder how it looks in VSCode, compared to viewing here on github

Reply:
the tabbing was a bit weird, I fixed it